main.c

```diff
@@ -354,8 +354,34 @@ void move_bug(
     struct board_tile board[SIZE][SIZE],
     struct bug_node* bug_linked_list
 ) {
-    printf("// TODO move_bug\n");
-    // Hint: use bug_linked_list to traverse over the list of bug
+    struct bug_node* current_bug = bug_linked_list;
+    while (current_bug != NULL) {
+        int next_col = current_bug->col;
+        int direction = board[current_bug->row][current_bug->col].type == WATER ? 1 : -1; // move right by default
```

Comment:
the current column of the bug should not be water right?
it's either turtle or log

probably what you want is either:
1. next column is not water, so move right, cmiiw?
```suggestion
        int direction = board[current_bug->row][current_bug->col+1].type != WATER ? 1 : -1; // move right by default
```
2. left column is water, so move right, but this will be false-positive if left-ward is log or turtle, and it will move to left
```suggestion
        int direction = board[current_bug->row][current_bug->col-1].type == WATER ? 1 : -1; // move right by default
```


and if you want by default moving right, maybe immediately set to `1`?